ipatests/test_integration/test_acme.py

```diff
@@ -0,0 +1,132 @@
+#
+# Copyright (C) 2020  FreeIPA Contributors see COPYING for license
+#
+
+import time
+
+import pytest
+
+from ipalib.constants import IPA_CA_RECORD
+from ipatests.test_integration.base import IntegrationTest
+from ipatests.pytest_ipa.integration import tasks
+from ipaplatform.osinfo import osinfo
+
+
+# RHEL does not have certbot.  EPEL's version is broken with
+# python-cryptography-2.3; likewise recent PyPI releases.
+# So for now, on RHEL we suppress tests that use certbot.
+skip_certbot_tests = osinfo.id not in ['fedora',]
+
+
+class TestACME(IntegrationTest):
+    """
+    Test the FreeIPA ACME service by using ACME clients on a FreeIPA client.
+
+    Right now the only thing we test is the Certbot client using
+    http-01 challenge with Certbot's standalone HTTP server.
+    We can add tests for DNS challenges later.
+
+    """
+
+    num_replicas = 0
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        # cache the acme service uri
+        acme_host = f'{IPA_CA_RECORD}.{cls.master.domain.name}'
+        cls.acme_server = f'https://{acme_host}/acme/directory'
+
+        # install packages before client install in case of IPA DNS problems
+        if not skip_certbot_tests:
+            cls.clients[0].run_command(['dnf', 'install', '-y', 'certbot'])
+        cls.clients[0].run_command(['dnf', 'install', '-y', 'mod_md'])
+
+        tasks.install_master(cls.master, setup_dns=True)
+
+        tasks.install_client(cls.master, cls.clients[0])
+        tasks.config_host_resolvconf_with_master_data(
+            cls.master, cls.clients[0]
+        )
+
+    #####################
+    # Enable ACME service
+    #####################
+
+    def test_acme_service_not_yet_enabled(self):
+        # --fail makes curl exit code 22 when response status >= 400.
+        # ACME service should return 503 because it was not enabled yet.
+        self.clients[0].run_command(
+            ['curl', '--fail', self.acme_server],
+            ok_returncode=22,
+        )
+
+    def test_enable_acme_service(self):
+        self.master.run_command(['ipa-acme-manage', 'enable'])
+
+        # wait a short time for Dogtag ACME service to observe config
+        # change and reconfigure itself to service requests
+        time.sleep(3)
+
+        # should succeed now
+        self.clients[0].run_command(['curl', '--fail', self.acme_server])
```

Comment:
A loop of several (10?) sleep/curl may be a more robust approach.

Reply:
I'm opting for 5 iters, 2 second sleep.  Updated in next push.